Bluetooth: host: Split SSP to its own module

joerchan opened the pull request:
Refactor the BR/EDR SSP feature out into it's own source file.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
Bluetooth: host: Split SSP to it's own module

Refactor the BR/EDR SSP feature out into it's own source file.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan commented on subsys/bluetooth/host/hci_core.c:
```diff
@@ -5208,23 +4872,24 @@ static const struct event_handler normal_events[] = {
 		      sizeof(struct bt_hci_evt_conn_request)),
 	EVENT_HANDLER(BT_HCI_EVT_CONN_COMPLETE, conn_complete,
```
@jhedberg I was thinking of continuing the pattern below to move other parts out, for example conn_complete here can be moved to conn.c as well as other conn related events. And adv/scan can be similarly extracted, although I would like periodic advertising in first. Do you see this as a good way forward?

jhedberg replied:
Sure, sounds like a good plan

carlescufi replied:
Yes, this makes sense, +1 to that.

joerchan committed:
Bluetooth: host: Move reset_pairing call to bt_conn_security_changed

Move call to reset_pairing to bt_conn_security_changed, since this
should always be done at the same time, and should be done in the same
order for all calls. For auth_complete this was reset after the
callbacks, which is inconsistent with the other events.

Also call it even if not connected anymore, the disconnection complete
event will be delivered after this event. Callback should alway be
called if security was initiated.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan commented:
DNM: Needs to be updated and merged after https://github.com/zephyrproject-rtos/zephyr/pull/27243

carlescufi reviewed (approved):
Fine with this, LGTM

joerchan commented:
> DNM: Needs to be updated and merged after #27243

Rebased.

joerchan commented:
@jhedberg I made same changes, would you mind looking at it one more time?
